File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.js
function solution(board) {
    let side = 0;
    let n = board.length;
    let m = board[0].length;

    if (n <= 1 || m <= 1) return 1;

    for (let i = 1; i < n; i++) {
        for (let j = 1; j < m; j++) {
            if (board[i][j] >= 1){
                let min = Math.min(board[i-1][j], board[i-1][j-1], board[i][j-1]);
                board[i][j] = min+1;
                
                side = Math.max(side, min+1);
            }
        }
    }

    return side**2;
}

File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.py
def solution(board):
    r,c = len(board), len(board[0])
    
    for i in range(1,r):
        for j in range(1,c):
            if board[i][j] == 1:
                up, left, up_left = (
                    board[i-1][j],
                    board[i][j-1],
                    board[i-1][j-1]
                )
                
                board[i][j] = min(up, left, up_left) + 1

    return max(max(r) for r in board) ** 2